src/ostree/ot-builtin-fsck.c

```diff
@@ -127,7 +127,7 @@ fsck_one_object (OstreeRepo            *repo,
                   if ((state & OSTREE_REPO_COMMIT_STATE_PARTIAL) == 0)
                     {
                       g_printerr ("Marking commit as partial: %s\n", parent_commit);
-                      if (!ostree_repo_mark_commit_partial (repo, parent_commit, TRUE, error))
+                      if (!ostree_repo_mark_commit_partial_reason (repo, parent_commit, TRUE, OSTREE_REPO_COMMIT_STATE_FSCK_PARTIAL, error))
```

Comment:
This is fine, but I just noticed that there's a race condition here - if `fsck` is interrupted between deleting the first object and marking the commit partial, we'd have a partial commit without the flag.

We should probably mark all commits partial *before* deletion.

(And it looks like we're marking the commit partial each time, which is a bit ugly, but not a new bug)